app/certificates.py

```diff
@@ -0,0 +1,158 @@
+        cert_handler.remove_files()
+        logger.info("The certificate was removed at %s." % certificate_folder)
+    else:
+        logger.info("No certificate was found on disk at %s." % certificate_folder)
+
+
+def __create_certificate(certificate_folder):
+    """
+    Ensures that the certificates are created and registered. If something is amiss, then the
+    configuration is fixed.
+
+    :param certificate_folder: Folder where the certificates are stored.
+    """
+
+    cert_handler = tk_framework_desktopserver.get_certificate_handler(certificate_folder)
+
+    # We only warn once.
+    warned = False
+    # Make sure the certificates exist.
+    if not cert_handler.exists():
+        logger.debug("Certificate doesn't exist on disk.")
+        # Start by unregistering certificates from the keychains, this can happen if the user
+        # wiped his certificates folder.
+        if cert_handler.is_registered():
+            # Warn once.
+            __warn_for_prompt()
+            logger.debug("Unregistering dangling certificate from database...")
+            warned = True
+            cert_handler.unregister()
+            logger.debug("Done.")
+        # Create the certificate files
+        logger.debug("About to create the certificates...")
+        cert_handler.create()
+        logger.info("Certificate created at %s." % certificate_folder)
+    else:
+        logger.info("Certificate already exist on disk at %s." % certificate_folder)
+
+    # Check if the certificates are registered with the keychain.
+    if not cert_handler.is_registered():
+        logger.debug("Certificate is not currently registered in the keychain.")
+        # Only if we've never been warned before.
+        if not warned:
+            __warn_for_prompt()
+        cert_handler.register()
+        logger.info("Certificate is now registered .")
+    else:
+        logger.info("Certificate is already registered.")
+
+
+if __name__ == '__main__':
+    # Add the modules files to PYTHOHPATH
+    sys.path.insert(0, "../python")
+    import tk_framework_desktopserver
+
+    # Make sure logger output goes to stdout
+    logger = tk_framework_desktopserver.get_logger()
+    logger.addHandler(logging.StreamHandler())
+
+    parser = optparse.OptionParser()
+    parser.add_option(
+        "--debug", action="store_true", default=False,
+        help="prints debugging message from the certificate generation."
+    )
+    parser.add_option(
+        "--remove", action="store_true", default=False,
+        help="prints debugging message from the certificate generation."
+    )
+
+    options, _ = parser.parse_args()
+
+    if options.debug is True:
+        logger.setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+
+    folder = os.environ.get("SGTK_BROWSER_INTEGRATION_CERTIFICATE", "../resources/keys")
+    folder = os.path.expanduser(folder)
+    folder = os.path.abspath(folder)
+    folder = os.path.normpath(folder)
```

Comment:
Along with the env vars above, should this be a path to a config.ini that the server would also point to so that you don't need different mechanisms for pointing this to a cert and the server itself?